src/teaching-element/Assessment/MultipleChoice.vue

```diff
@@ -3,58 +3,76 @@
     <span class="form-label">Solution</span>
     <ul class="answers">
       <li
-        v-for="(answer, index) in answers"
-        :key="index"
-        :class="{ selected: isSelected(index) }">
+        v-for="(answer, index) in mappedAnswers"
+        :key="answer.id"
+        :class="getAnswerClass(index)">
         <input
           v-model="userAnswer"
-          :value="index"
+          :id="answer.id"
           :disabled="disabled"
+          :value="index"
           @change="update"
           class="answers-checkbox"
           type="checkbox">
-        <span class="order">{{ transform(index) }}.</span>
-        <span>{{ answer }}</span>
+        <label :for="answer.id">{{ transform(index) }}. {{ answer.answer }}</label>
       </li>
     </ul>
   </div>
 </template>
 
 <script>
-import includes from 'lodash/includes';
-import isEmpty from 'lodash/isEmpty';
+import cuid from 'cuid';
 import { rules } from '../../util/listingType';
 
 const defaults = { type: 'upper-latin' };
 
 export default {
   props: {
     answers: { type: Array, required: true },
+    correct: { type: Array, required: true },
     disabled: { type: Boolean, default: false },
     options: { type: Object, default: () => ({}) },
     retake: { type: Boolean, default: false },
     submission: { type: Array, default: () => ([]) }
   },
   data() {
-    return { userAnswer: this.submission || [] };
+    return {
+      userAnswer: this.submission || []
+    };
   },
   computed: {
+    mappedAnswers() {
+      return this.answers.map((answer, index) => ({
+        answer,
+        correct: this.correct.includes(index),
+        id: cuid()
+      }));
+    },
     type() {
       const options = this.options.multipleChoice || defaults;
       return options.type;
     }
   },
   methods: {
-    update() {
-      let { userAnswer } = this;
-      userAnswer = isEmpty(userAnswer) ? null : userAnswer.sort();
-      this.$emit('update', { userAnswer });
+    getAnswerClass(index) {
+      const selected = this.userAnswer.includes(index) ? 'selected' : '';
+      if (!this.disabled || !this.options.enableHighlighting) return [selected];
+      return [
+        selected,
+        this.isAnswerCorrect(index) ? 'te-correct' : 'te-incorrect'
+      ];
     },
-    isSelected(index) {
-      return includes(this.userAnswer, index);
+    isAnswerCorrect(index) {
+      const { userAnswer } = this;
+      return !(userAnswer.includes(index) ^ this.mappedAnswers[index].correct);
     },
     transform(index) {
       return rules[this.type](index);
+    },
+    update() {
+      let { userAnswer } = this;
+      userAnswer = userAnswer.length > 0 ? userAnswer.sort() : null;
```

Comment:
Maybe put `userAnswer.length` instead of `userAnswer.length > 0`?

Reply:
👍